src/OagBundle/Service/ChangeService.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace OagBundle\Service;
+
+use OagBundle\Entity\Change;
+
+/**
+ * A service for dealing with Change entities. Generally, this involves applying
+ * them to an IATI file.
+ */
+class ChangeService extends AbstractService {
+
+    /**
+     * Summarise the history of an activity in the application and present it
+     * as an applicable change for the current state of the activity given.
+     *
+     * This is useful for working out what past changes can be reapplied to
+     * activities in an untouched IATI file, for example.
+     *
+     * @param \SimpleXMLElement &$activity the activity to find the history of
+     * @return Change
+     */
+    public function summariseHistory(\SimpleXMLElement &$activity) {
```

Comment:
why does $activity need to be passed in byref here?

Reply:
Probably doesn't. Might remove, as it's easier to change back later if need be.